pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -82,9 +123,11 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	// Retry failures are treated the same as Abort.
 	switch hook.FailurePolicy {
 	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
-		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
+		e.event(deployment, kapi.EventTypeWarning, "Failed", fmt.Sprintf("The %s hook failed: %v, aborting deployment %s/%s", label, err, deployment.Namespace, deployment.Name))
```

Comment:
I'm not sure.  What confidence do we have that the events actually
convey the right info?  Certainly any sort of admission / quota /
policy failures *must* be reflected.  I'd almost prefer to start
there, and we don't need events for that to be correct (well, except
for the hooks).